### PR TITLE
Fixes #7911: Add an output to all Rudder commands

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/7911

### CODE REVIEW

**Comment by amousset** [comment]
PR updated

File: share/commands/agent-reinit [hunk 1]
```diff
@@ -27,5 +27,7 @@ rm -rf /opt/rudder/var/fusioninventory/*
 # - reset cfengine status
 "${BASEDIR}/agent-reset"
 
+printf "${GREEN}ok${NORMAL}: Rudder agent state has been reset. The keys and uuid were replaced.\nNow trigerring an inventory..."
```
Comment (peckpeck) [inline share/commands/agent-reinit]: The first sentence is already printed one line above, just remove it.

File: share/commands/agent-update [hunk 2]
```diff
@@ -34,3 +34,10 @@ while getopts "iIvdic" opt; do
 done
 
 /var/rudder/cfengine-community/bin/cf-agent ${VERBOSITY} ${COLOR} -K -f failsafe.cf
+
+if [ $? -eq 0 ]; then
+  printf "${GREEN}ok${NORMAL}: Rudder agent promises were updated.\n"
+else
+  printf "${RED}error${NORMAL}: Rudder agent promises could not be updated.\n" 1>&2
+  exit 1
```
Comment (peckpeck) [inline share/commands/agent-update]: Isn't there any other code than 1 ?
I think we should store and reuse $? here

File: share/commands/remote-run [hunk 3]
```diff
@@ -20,17 +20,18 @@
...
     v)
-      VERBOSITY="-v"
+      VERBOSITY="-v ${DEBUG_CLASS}"
       ;;
     d)
-      VERBOSITY="-d"
+      VERBOSITY="-d ${DEBUG_CLASS}"
       ;;
     c)
       COLOR=""
```
Comment (peckpeck) [inline share/commands/remote-run]: Since you're here, should you print something after cf-runagen call ?

**Comment by amousset** [comment]
PR updated

File: share/commands/agent-check [hunk 4]
```diff
@@ -7,4 +7,32 @@
...
+done
+
 /opt/rudder/bin/check-rudder-agent
+
+if [ $? -eq 0 ]; then
+  [ "$QUIET" = false ] && printf "${GREEN}ok${NORMAL}: Rudder agent check ran without errors.\n"
+else
+  printf "${RED}error${NORMAL}: Rudder agent check failed.\n" 1>&2
+  exit 1
+fi
```
Comment (peckpeck) [inline share/commands/agent-check]: Missing eol
Isn't your editor vi ?

File: share/lib/common.sh [hunk 5]
```diff
@@ -31,3 +31,6 @@ if [ -t 1 ]; then
 else
     clear_colors
 fi
+
+# Paths
+$RUDDER_VAR="/var/rudder"
```
Comment (peckpeck) [inline share/lib/common.sh]: No eol

File: share/commands/agent-check [hunk 6]
```diff
@@ -7,4 +7,32 @@
...
+# @man +
+# @man *Options*:
+# @man +
+# @man *-q*: run the agent in quiet mode (display only error messages)
+# @man +
+# @man *-c*: run the agent without color output
+
+. "${BASEDIR}/../lib/common.sh"
+
+QUIET=false
```
Comment (peckpeck) [inline share/commands/agent-check]: Even if the option is named "quiet" to denote the default value of "not quiet", always use positive variables; ie VERBOSE.
Otherwise you end up using negation everywhere (if not quiet) and this makes the code unnecessarily hard to read.

So $QUIET -> $VERBOSE

File: share/commands/agent-reinit [hunk 7]
```diff
@@ -22,10 +63,13 @@ else
...
 rm -rf /opt/rudder/var/fusioninventory/*
 
 # - reset cfengine status
 "${BASEDIR}/agent-reset"
 
+[ "$VERBOSE" = true ] && echo "Triggering an inventory..."
+
 # - resend inventory
-"${BASEDIR}/agent-inventory"
+"${BASEDIR}/agent-inventory" $@
```
Comment (peckpeck) [inline share/commands/agent-reinit]: always use quotes around $@

**Comment by peckpeck** [comment]
ping @amousset

**Comment by amousset** [comment]
PR updated

File: share/commands/agent-check (hunk 6 above)
Comment (amousset) [inline share/commands/agent-check]: Currently
* QUIET = less verbose than default
* VERBOSE = more verbose than default

I would prefer avoid using VERBOSE for this. I could be DISPLAY_OUTPUT, or something better.

File: share/commands/remote-run (hunk 3 above)
Comment (amousset) [inline share/commands/remote-run]: We could also modify the agent-run wrapper to be able to parse remote-run output. I don't know if it is worth it, what do you think?

Comment (peckpeck) [inline share/commands/remote-run]: Yes it could be cool if you could do it.
Is the output for remote agent the same as for a local agent ?

Comment (amousset) [inline share/commands/remote-run]: Almost:
```
# rudder remote run 192.168.46.3
192.168.46.3> cf-serverd executing cfruncommand: /var/rudder/cfengine-community/bin/cf-agent -f failsafe.cf && /var/rudder/cfengine-community/bin/cf-agent -I -Dcfruncommand
192.168.46.3> -> rudder     info: Error reading assumed integer value '0.000000' => 'non-value', found remainder '000000'
192.168.46.3> -> rudder     info: Error reading assumed integer value '0.000000' => 'non-value', found remainder '000000'
192.168.46.3> -> rudder     info: Error reading assumed integer value '0.000000' => 'non-value', found remainder '000000'
192.168.46.3> -> rudder     info: Error reading assumed integer value '0.000000' => 'non-value', found remainder '000000'
192.168.46.3> -> rudder     info: Error reading assumed integer value '0.000000' => 'non-value', found remainder '000000'
192.168.46.3> -> rudder     info: Error reading assumed integer value '0.000000' => 'non-value', found remainder '000000'
192.168.46.3> -> R: @@Common@@log_info@@hasPolicyServer-root@@common-root@@4@@common@@StartRun@@2016-02-15 10:12:39+00:00##a568d40a-9040-4cf3-a291-04f10e375ec1@#Start execution with config [-1778907046]
192.168.46.3> -> rudder     info: Automatically promoting context scope for 'heartbeat_sent' to namespace visibility, due to persistence
192.168.46.3> -> R: @@Common@@result_success@@hasPolicyServer-root@@common-root@@4@@Process checking@@None@@2016-02-15 10:12:39+00:00##a568d40a-9040-4cf3-a291-04f10e375ec1@#There is an acceptable number of CFEngine processes running on the machine
192.168.46.3> -> R: @@Common@@result_success@@hasPolicyServer-root@@common-root@@4@@CRON Daemon@@None@@2016-02-15 10:12:39+00:00##a568d40a-9040-4cf3-a291-04f10e375ec1@#Cron daemon status was correct
192.168.46.3> -> R: @@Common@@result_success@@hasPolicyServer-root@@common-root@@4@@Binaries update@@None@@2016-02-15 10:12:39+00:00##a568d40a-9040-4cf3-a291-04f10e375ec1@#The CFengine binaries in /var/rudder/cfengine-community/bin are up to date
192.168.46.3> -> R: @@Common@@log_info@@hasPolicyServer-root@@common-root@@4@@Log system for reports@@None@@2016-02-15 10:12:39+00:00##a568d40a-9040-4cf3-a291-04f10e375ec1@#Detected running syslog as rsyslog
192.168.46.3> -> R: @@Common@@result_success@@hasPolicyServer-root@@common-root@@4@@Log system for reports@@None@@2016-02-15 10:12:39+00:00##a568d40a-9040-4cf3-a291-04f10e375ec1@#Logging system for report centralization is already correctly configured
192.168.46.3> -> rudder     info: Executing 'no timeout' ... '/var/rudder/tools/check-rsyslog-version'
192.168.46.3> -> rudder     info: Completed execution of '/var/rudder/tools/check-rsyslog-version'
192.168.46.3> -> rudder     info: Executing 'no timeout' ... '/usr/bin/curl -L -k -1 -s -f --proxy '' -o "/var/rudder/cfengine-community/rudder-server-uuid.txt" https://server.rudder.local/uuid'
192.168.46.3> -> rudder     info: Completed execution of '/usr/bin/curl -L -k -1 -s -f --proxy '' -o "/var/rudder/cfengine-community/rudder-server-uuid.txt" https://server.rudder.local/uuid'
192.168.46.3> -> R: @@Inventory@@result_success@@inventory-all@@inventory-all@@4@@inventory@@None@@2016-02-15 10:12:39+00:00##a568d40a-9040-4cf3-a291-04f10e375ec1@#Next inventory scheduled between 00:00 and 06:00
192.168.46.3> -> R: @@checkGenericFileContent@@result_success@@32377fd7-02fd-43d0-aab7-28460a91347b@@0591b431-bcb1-422c-966f-1c8ff54b5444@@15@@File@@/tmp/${toto.toto}@@2016-02-15 10:12:39+00:00##a568d40a-9040-4cf3-a291-04f10e375ec1@#The file /tmp/${toto.toto} was already in accordance with the policy
192.168.46.3> -> R: @@checkGenericFileContent@@result_success@@32377fd7-02fd-43d0-aab7-28460a91347b@@0591b431-bcb1-422c-966f-1c8ff54b5444@@15@@Line deletion regular expressions@@/tmp/${toto.toto}@@2016-02-15 10:12:39+00:00##a568d40a-9040-4cf3-a291-04f10e375ec1@#The file /tmp/${toto.toto} was not set for any line deletion
192.168.46.3> -> R: @@checkGenericFileContent@@result_success@@32377fd7-02fd-43d0-aab7-28460a91347b@@0591b431-bcb1-422c-966f-1c8ff54b5444@@15@@Line replacement regular expressions@@/tmp/${toto.toto}@@2016-02-15 10:12:39+00:00##a568d40a-9040-4cf3-a291-04f10e375ec1@#The file /tmp/${toto.toto} was not set for any line replacement
192.168.46.3> -> R: @@checkGenericFileContent@@result_success@@32377fd7-02fd-43d0-aab7-28460a91347b@@0591b431-bcb1-422c-966f-1c8ff54b5444@@15@@Permission adjustment@@/tmp/${toto.toto}@@2016-02-15 10:12:39+00:00##a568d40a-9040-4cf3-a291-04f10e375ec1@#The file /tmp/${toto.toto} uses default permissions
192.168.46.3> -> R: @@checkGenericFileContent@@result_success@@32377fd7-02fd-43d0-aab7-28460a91347b@@0591b431-bcb1-422c-966f-1c8ff54b5444@@15@@Enforce content by section@@/tmp/${toto.toto}@@2016-02-15 10:12:39+00:00##a568d40a-9040-4cf3-a291-04f10e375ec1@#The file /tmp/${toto.toto} was not set for section edition, skipping
192.168.46.3> -> R: @@checkGenericFileContent@@result_success@@32377fd7-02fd-43d0-aab7-28460a91347b@@0591b431-bcb1-422c-966f-1c8ff54b5444@@15@@Post-modification hook@@/tmp/${toto.toto}@@2016-02-15 10:12:39+00:00##a568d40a-9040-4cf3-a291-04f10e375ec1@#No command for /tmp/${toto.toto} was to be executed
192.168.46.3> -> R: @@genericVariableDefinition@@result_success@@32377fd7-02fd-43d0-aab7-28460a91347b@@a95a1769-b3bc-4f76-aaa9-7615f383221c@@15@@Variable definition@@pedro2@@2016-02-15 10:12:39+00:00##a568d40a-9040-4cf3-a291-04f10e375ec1@#A generic variable definition is set for variable pedro2
192.168.46.3> -> R: @@template@@result_success@@32377fd7-02fd-43d0-aab7-28460a91347b@@3f3aaf4d-549b-4072-ab56-ec9b75ff9cff@@15@@Variable string@@vari@@2016-02-15 10:12:39+00:00##a568d40a-9040-4cf3-a291-04f10e375ec1@#Set the string pref.vari to value plop was correct
192.168.46.3> -> R: @@template@@result_success@@32377fd7-02fd-43d0-aab7-28460a91347b@@3f3aaf4d-549b-4072-ab56-ec9b75ff9cff@@15@@Directory create@@/tmp/plop@@2016-02-15 10:12:39+00:00##a568d40a-9040-4cf3-a291-04f10e375ec1@#Create directory /tmp/plop was correct
192.168.46.3> -> R: [ERROR] Promise could not be repaired, error encountered: Build file /tmp/output from mustache template /tmp/template
192.168.46.3> -> R: @@template@@result_error@@32377fd7-02fd-43d0-aab7-28460a91347b@@3f3aaf4d-549b-4072-ab56-ec9b75ff9cff@@15@@File from a mustache template@@/tmp/output@@2016-02-15 10:12:39+00:00##a568d40a-9040-4cf3-a291-04f10e375ec1@#Build file /tmp/output from mustache template /tmp/template could not be repaired
192.168.46.3> -> R: @@template@@result_success@@32377fd7-02fd-43d0-aab7-28460a91347b@@3f3aaf4d-549b-4072-ab56-ec9b75ff9cff@@15@@File create@@/tmp/pouet@@2016-02-15 10:12:39+00:00##a568d40a-9040-4cf3-a291-04f10e375ec1@#Create file /tmp/pouet was correct
192.168.46.3> -> R: @@template@@result_success@@32377fd7-02fd-43d0-aab7-28460a91347b@@3f3aaf4d-549b-4072-ab56-ec9b75ff9cff@@15@@Variable string@@toto@@2016-02-15 10:12:39+00:00##a568d40a-9040-4cf3-a291-04f10e375ec1@#Set the string toto.toto to value puc was correct
192.168.46.3> -> R: @@Common@@log_info@@hasPolicyServer-root@@common-root@@4@@common@@EndRun@@2016-02-15 10:12:39+00:00##a568d40a-9040-4cf3-a291-04f10e375ec1@#End execution with config [-1778907046]
root@server:/tmp# rudder remote run 192.168.46.3
192.168.46.3> cf-serverd executing cfruncommand: /var/rudder/cfengine-community/bin/cf-agent -f failsafe.cf && /var/rudder/cfengine-community/bin/cf-agent -I -Dcfruncommand
192.168.46.3> -> rudder     info: Error reading assumed integer value '0.000000' => 'non-value', found remainder '000000'
192.168.46.3> -> rudder     info: Error reading assumed integer value '0.000000' => 'non-value', found remainder '000000'
192.168.46.3> -> rudder     info: Error reading assumed integer value '0.000000' => 'non-value', found remainder '000000'
192.168.46.3> -> rudder     info: Error reading assumed integer value '0.000000' => 'non-value', found remainder '000000'
192.168.46.3> -> rudder     info: Error reading assumed integer value '0.000000' => 'non-value', found remainder '000000'
192.168.46.3> -> rudder     info: Error reading assumed integer value '0.000000' => 'non-value', found remainder '000000'
192.168.46.3> -> R: @@Common@@log_info@@hasPolicyServer-root@@common-root@@4@@common@@StartRun@@2016-02-15 10:12:44+00:00##a568d40a-9040-4cf3-a291-04f10e375ec1@#Start execution with config [-1778907046]
192.168.46.3> -> rudder     info: Automatically promoting context scope for 'heartbeat_sent' to namespace visibility, due to persistence
192.168.46.3> -> R: @@Common@@log_info@@hasPolicyServer-root@@common-root@@4@@common@@EndRun@@2016-02-15 10:12:44+00:00##a568d40a-9040-4cf3-a291-04f10e375ec1@#End execution with config [-1778907046]

```

Comment (peckpeck) [inline share/commands/remote-run]: The go on, but you should do it in a separate PR

Comment (amousset) [inline share/commands/remote-run]: http://www.rudder-project.org/redmine/issues/7945

**Comment by peckpeck** [comment]
OK, merging
